tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2017 comsuisse AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @addtogroup t_i2s_speed
+ * @{
+ * @defgroup t_i2s_speed test_i2s_speed
+ * @brief TestPurpose: verify maximum speed the I2S driver can support
+ * @}
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <i2s.h>
+
+#define I2S_DEV_NAME "I2S_0"
+#define NUM_BLOCKS 20
+#define SAMPLE_NO 64
+
+/* The data_l represent a sine wave */
+static s16_t data_l[SAMPLE_NO] = {
+	  3211,   6392,   9511,  12539,  15446,  18204,  20787,  23169,
+	 25329,  27244,  28897,  30272,  31356,  32137,  32609,  32767,
+	 32609,  32137,  31356,  30272,  28897,  27244,  25329,  23169,
+	 20787,  18204,  15446,  12539,   9511,   6392,   3211,      0,
+	 -3212,  -6393,  -9512, -12540, -15447, -18205, -20788, -23170,
+	-25330, -27245, -28898, -30273, -31357, -32138, -32610, -32767,
+	-32610, -32138, -31357, -30273, -28898, -27245, -25330, -23170,
+	-20788, -18205, -15447, -12540,  -9512,  -6393,  -3212,     -1,
+};
+
+/* The data_r represent a sine wave with double the frequency of data_l */
+static s16_t data_r[SAMPLE_NO] = {
+	 32609,  32137,  31356,  30272,  28897,  27244,  25329,  23169,
+	 20787,  18204,  15446,  12539,   9511,   6392,   3211,      0,
+	 -3212,  -6393,  -9512, -12540, -15447, -18205, -20788, -23170,
+	-25330, -27245, -28898, -30273, -31357, -32138, -32610, -32767,
+	-32610, -32138, -31357, -30273, -28898, -27245, -25330, -23170,
+	-20788, -18205, -15447, -12540,  -9512,  -6393,  -3212,     -1,
+	  3211,   6392,   9511,  12539,  15446,  18204,  20787,  23169,
+	 25329,  27244,  28897,  30272,  31356,  32137,  32609,  32767,
+};
+
+#define BLOCK_SIZE (2 * sizeof(data_l))
+
+K_MEM_SLAB_DEFINE(rx_0_mem_slab, BLOCK_SIZE, NUM_BLOCKS, 1);
+K_MEM_SLAB_DEFINE(tx_0_mem_slab, BLOCK_SIZE, NUM_BLOCKS, 1);
+
+static void fill_buf(s16_t *tx_block, int att)
+{
+	for (int i = 0; i < SAMPLE_NO; i++) {
+		tx_block[2 * i] = data_l[i] >> att;
+		tx_block[2 * i + 1] = data_r[i] >> att;
+	}
+}
+
+static int verify_buf(s16_t *rx_block, int att)
+{
+	for (int i = 0; i < SAMPLE_NO; i++) {
+		if (rx_block[2 * i] != data_l[i] >> att) {
+			TC_PRINT("Error: att %d: data_l mismatch at position "
+				 "%d, expected %d, actual %d\n",
+				 att, i, data_l[i] >> att, rx_block[2 * i]);
+			return -TC_FAIL;
+		}
+		if (rx_block[2 * i + 1] != data_r[i] >> att) {
+			TC_PRINT("Error: att %d: data_r mismatch at position "
+				 "%d, expected %d, actual %d\n",
+				 att, i, data_r[i] >> att, rx_block[2 * i + 1]);
+			return -TC_FAIL;
+		}
+	}
+
+	return TC_PASS;
+}
+
+#define TIMEOUT          2000
+#define FRAME_CLK_FREQ   48000
```

Comment:
Maximum interface speed should be a parameter which is defined independently for every SPI driver / board and not hard coded in a .c file. I did not found however a good example of how to define such testcase parameters. Shall I use some Kconfig test option, e.g. TEST_I2S_FRAME_CLK_FREQ_MAX? If yes, where should I set its value?